Unicode UI/MitSciUnicodeUI.pax

```diff
@@ -0,0 +1,1283 @@
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Models\List\Dolphin List Models';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Presenters\List\Dolphin List Presenter';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Base\Dolphin MVP Base';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Presenters\Prompters\Dolphin Prompter';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Presenters\Text\Dolphin Text Presenter';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Type Converters\Dolphin Type Converters';
+	add: '..\..\Core\Object Arts\Dolphin\MVP\Models\Value\Dolphin Value Models';
+	add: '..\..\Core\Contributions\Refactory\Refactoring Browser\Environments\RBEnvironments';
+	add: '..\..\Core\Object Arts\Dolphin\System\Compiler\Smalltalk Parser';
+	add: '..\Udo Schneider\Unicode\US Unicode';
+	yourself).
+
+package!
+
+"Class Definitions"!
+
+Object subclass: #MitSciUnicodeAPIScanner
+	instanceVariableNames: 'ansiAPIFunctionMethods'
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+ODBCLibrary subclass: #ODBCLibraryW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+GDILibrary subclass: #GDILibraryW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+UserLibrary subclass: #UserLibraryW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+WNDCLASS subclass: #WNDCLASSW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+MitSciUnicodeAPIScanner subclass: #MitSciUnicodeTransformer
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+Shell subclass: #TestUnicodeShell
+	instanceVariableNames: 'listPresenter textPresenter'
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+FlipperInspector subclass: #FlipperInspectorW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+TestUnicodeShell subclass: #TestNonUnicodeShell
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+
+"Global Aliases"!
+
+
+"Loose Methods"!
+
+!ByteArray methodsFor!
+
+asUnicodeString
+	"Answer a UnicodeString containing the same elements as the receiver."
+	"cdemers 12/7/2016 Added for Unicode support."
+
+	^UnicodeString fromAddress: self yourAddress length: self size / 2! !
+!ByteArray categoriesFor: #asUnicodeString!converting!public! !
+
+!Canvas methodsFor!
+
+gdiLibrary
+
+	"cdemers 12/6/2016  I don't presently have a good way to default this.  Going to try to override based on string type arguments wherever I can."
+	^GDILibrary default
+!
+
+gdiLibraryForString: aStringOrUnicodeString 
+
+	"cdemers 12/6/2016"
+	^((aStringOrUnicodeString class == UnicodeString ) ifTrue: [GDILibraryW] ifFalse: [GDILibrary]) default
```

Comment:
Would it make sense to just ask the string? To avoid adding a method per library to each of the different string classes, this could be done by passing the library class to a string method, and then having the string ask the library class for the correct library instance, e.g.
```
...
lib := aString libraryInstance: GDILibrary.
lib blah: aString.
...
String>>libraryInstance: anExternalLibrary
   ^anExternalLibrary default
...
UnicodeString>>libraryInstance: anExternalLibrary
   ^anExternalLibrary wide
```
Would that work?